sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py

```diff
@@ -360,3 +361,48 @@ async def negotiate(self):
         if returned_header == TLS_HEADER_FRAME:
             raise ValueError("Mismatching TLS header protocol. Excpected: {}, received: {}".format(
                 TLS_HEADER_FRAME, returned_header[1]))
+
+
+class WebSocketTransportAsync(WebSocketTransport):
+    async def _read(self, n, buffer=None, **kwargs): # pylint: disable=unused-arguments
+        """Read exactly n bytes from the peer."""
+        
+        length = 0
+        view = buffer or memoryview(bytearray(n))
+        nbytes = self._read_buffer.readinto(view)
+        length += nbytes
+        n -= nbytes
+        while n:
+            data = await asyncio.get_event_loop().run_in_executor(
+                None, self.ws.recv
+                )
+
+            if len(data) <= n:
+                view[length: length + len(data)] = data
+                n -= len(data)
+            else:
+                view[length: length + n] = data[0:n]
+                self._read_buffer = BytesIO(data[n:])
+                n = 0
+        return view
+
+    async def _shutdown_transport(self):
+        """Do any preliminary work in shutting down the connection."""
+        await asyncio.get_event_loop().run_in_executor(
+                    None, self.ws.close
+                    )
+
+    async def _write(self, s):
+        """Completely write a string to the peer."""
+        try:
+            """
+            ABNF, OPCODE_BINARY = 0x2
+            See http://tools.ietf.org/html/rfc5234
+            http://tools.ietf.org/html/rfc6455#section-5.2
+            """
+            from websocket import ABNF
+            await asyncio.get_event_loop().run_in_executor(
+                    None, self.ws.send, s, ABNF.OPCODE_BINARY
+                    )
+        except ImportError:
+            raise ValueError("Please install websocket-client library to use websocket transport.")
```

Comment:
same comment, we could just place code "0x2" instead of importing each time in "write"